contracts/tokenlock/TokenLock.sol

```diff
@@ -190,17 +191,27 @@ abstract contract TokenLock is Ownable, ITokenLock {
         if (!revocable)
             revert LockIsNonRevocable();
 
-        uint256 unvestedAmount = managedAmount - vestedAmount();
+        if (isRevoked)
+            revert LockIsAlreadyRevoked();
+
+        uint256 vestedAmount = vestedAmount();
+
+        uint256 unvestedAmount = managedAmount - vestedAmount;
         if (unvestedAmount == 0)
             revert NoAvailableUnvestedAmount();
 
         isRevoked = true;
 
+        managedAmount = vestedAmount;
+
+        // solhint-disable-next-line not-rely-on-time
+        endTime = block.timestamp;
+
         token.safeTransfer(owner(), unvestedAmount);
 
         emit TokensRevoked(beneficiary, unvestedAmount);
 
-        selfdestruct(payable(msg.sender));
+        trySelfDestruct();
```

Comment:
perhaps we could consider just dropping the selfdestruct code, I am not sure why it is even thre except to get some amount of gas back (but we are also spending gas each time it is called without effect)